### PR TITLE
Move programs/system into runtime/

### LINKED ISSUE #3232: Circular dependency between runtime and programs/system
Can't publish to crates.io because runtime depends on programs/system depends on runtime.   We can merge the two crates to resolve.

### PR DESCRIPTION
**Pull request by mvines**
`runtime/` depends on `programs/system/` depends on `runtime/`.  Squash into one.

Fixes #3232

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -7,6 +7,7 @@ pub mod loader_utils;
 mod native_loader;
 pub mod runtime;
 mod status_cache;
+pub mod system_program;
```
Comment (garious) [inline runtime/src/lib.rs]: That needed to be pub?

Comment (garious) [inline runtime/src/lib.rs]: That needed to be pub?

Comment (mvines) [reply runtime/src/lib.rs]: nerp, copypasta

**Comment by mvines** [comment]
grumble `test_leader_failure_3` is flaky

**Pull request merged**